vulnerabilities/pipelines/v2_importers/github_osv_live_importer.py

```diff
@@ -0,0 +1,186 @@
+import json
+from typing import Iterable
+from typing import Optional
+
+import requests
+from packageurl import PackageURL
+from univers.version_range import RANGE_CLASS_BY_SCHEMES
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipelineV2
+
+
+class GithubOSVLiveImporterPipeline(VulnerableCodeBaseImporterPipelineV2):
+    """
+    GithubOSV Live Importer Pipeline
+
+    Collect advisories from GitHub Advisory Database for a single PURL.
+    """
+
+    pipeline_id = "github_osv_live_importer_v2"
+    spdx_license_expression = "CC-BY-4.0"
+    license_url = "https://github.com/github/advisory-database/blob/main/LICENSE.md"
+    supported_types = ["pypi", "npm", "maven", "composer", "hex", "gem", "nuget", "cargo"]
+
+    @classmethod
+    def steps(cls):
+        return (
+            cls.get_purl_inputs,
+            cls.collect_and_store_advisories,
+        )
+
+    def get_purl_inputs(self):
+        purl = self.inputs["purl"]
+        if not purl:
+            raise ValueError("PURL is required for GithubOSVLiveImporterPipeline")
+
+        if isinstance(purl, str):
+            purl = PackageURL.from_string(purl)
+
+        if not isinstance(purl, PackageURL):
+            raise ValueError(f"Object of type {type(purl)} {purl!r} is not a PackageURL instance")
+
+        if purl.type not in self.supported_types:
+            raise ValueError(
+                f"PURL: {purl!s} is not among the supported package types {self.supported_types!r}"
+            )
+
+        if not purl.version:
+            raise ValueError(f"PURL: {purl!s} is expected to have a version")
+
+        self.purl = purl
+
+    def advisories_count(self):
+        self.advisories = fetch_github_osv_advisories_for_purl(self.purl)
+        return len(self.advisories)
+
+    def collect_advisories(self) -> Iterable[AdvisoryData]:
+        from vulnerabilities.importers.osv import parse_advisory_data_v2
+
+        supported_ecosystems = [
+            "pypi",
+            "npm",
+            "maven",
+            # "golang",
+            "composer",
+            "hex",
+            "gem",
+            "nuget",
+            "cargo",
+        ]
+
+        input_version = self.purl.version
+        vrc = RANGE_CLASS_BY_SCHEMES[self.purl.type]
+        version_obj = vrc.version_class(input_version)
+
+        for adv in self.advisories:
+            adv_id = adv.get("id")
+            advisory_url = build_github_repo_advisory_url(adv, adv_id)
```

Comment:
Use consistent naming: choose either full names or abbreviations, but stay consistent
```suggestion
            adv_url = build_github_repo_advisory_url(adv, adv_id)
```